src/components/LazyList.jsx

```diff
@@ -19,9 +19,18 @@ const LazyList = ({
   hasMore,
   emptyListComponent,
   renderItem,
-  pageStart
+  pageStart,
+  minimumInitialLoading = 1250
+  // 1250 allows the animation to complete one full cycle.
 }) => {
   const isEmpty = !items.length;
+  const [initialLoading, setInitialLoading] = useState(true);
+  if (minimumInitialLoading > 0 && initialLoading) {
+    setTimeout(() => {
+      setInitialLoading(false);
+    }, minimumInitialLoading);
+  }
```

Comment:
I think you should wrap this into a `useEffect` to avoid re-renders and improve performance.

```suggestion
  useEffect(() => {
    if (minimumInitialLoading > 0 && initialLoading) {
      setTimeout(() => {
        setInitialLoading(false);
      }, minimumInitialLoading);
    }
  }, [minimumInitialLoading, initialLoading]);
```